internal/pkg/describe/backend_app.go

```diff
@@ -0,0 +1,215 @@
+				CPU:         appParams[stack.AppTaskCPUParamKey],
+				Memory:      appParams[stack.AppTaskMemoryParamKey],
+			})
+			backendAppEnvVars, err := d.appDescriber.EnvVars()
+			if err != nil {
+				return nil, fmt.Errorf("retrieve environment variables: %w", err)
+			}
+			envVars = append(envVars, flattenEnvVars(env.Name, backendAppEnvVars)...)
+			continue
+		}
+		if !IsStackNotExistsErr(err) {
+			return nil, fmt.Errorf("retrieve application deployment configuration: %w", err)
+		}
+	}
+	sort.SliceStable(envVars, func(i, j int) bool { return envVars[i].Environment < envVars[j].Environment })
+	sort.SliceStable(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
+
+	resources := make(map[string][]*CfnResource)
+	if d.enableResources {
+		for _, env := range environments {
+			stackResources, err := d.appDescriber.AppStackResources()
+			if err == nil {
+				resources[env.Name] = flattenResources(stackResources)
+				continue
+			}
+			if !IsStackNotExistsErr(err) {
+				return nil, fmt.Errorf("retrieve application resources: %w", err)
+			}
+		}
+	}
+
+	return &BackendAppDesc{
+		AppName:          d.app.Name,
+		Type:             d.app.Type,
+		Project:          d.app.Project,
+		Configurations:   configs,
+		ServiceDiscovery: services,
+		Variables:        envVars,
+		Resources:        resources,
+	}, nil
+}
+
+// BackendAppDesc contains serialized parameters for a backend application.
+type BackendAppDesc struct {
+	AppName          string                    `json:"appName"`
+	Type             string                    `json:"type"`
+	Project          string                    `json:"project"`
+	Configurations   []*AppConfig              `json:"configurations"`
+	ServiceDiscovery []*ServiceDiscovery       `json:"serviceDiscovery"`
+	Variables        []*EnvVars                `json:"variables"`
+	Resources        map[string][]*CfnResource `json:"resources,omitempty"`
+}
+
+// JSONString returns the stringified BackendApp struct with json format.
+func (w *BackendAppDesc) JSONString() (string, error) {
+	b, err := json.Marshal(w)
+	if err != nil {
+		return "", fmt.Errorf("marshal applications: %w", err)
+	}
+	return fmt.Sprintf("%s\n", b), nil
+}
+
+// HumanString returns the stringified BackendApp struct with human readable format.
+func (w *BackendAppDesc) HumanString() string {
+	var b bytes.Buffer
+	writer := tabwriter.NewWriter(&b, minCellWidth, tabWidth, cellPaddingWidth, paddingChar, noAdditionalFormatting)
+	fmt.Fprintf(writer, color.Bold.Sprint("About\n\n"))
+	writer.Flush()
+	fmt.Fprintf(writer, "  %s\t%s\n", "Project", w.Project)
+	fmt.Fprintf(writer, "  %s\t%s\n", "Name", w.AppName)
+	fmt.Fprintf(writer, "  %s\t%s\n", "Type", w.Type)
+	fmt.Fprintf(writer, color.Bold.Sprint("\nConfigurations\n\n"))
+	writer.Flush()
+	fmt.Fprintf(writer, "  %s\t%s\t%s\t%s\t%s\n", "Environment", "Tasks", "CPU (vCPU)", "Memory (MiB)", "Port")
+	for _, config := range w.Configurations {
```

Comment:
Since so many of these sections "Configurations", "Service Discovery", "Variables", "Resources" are going to be repeated across app types, do you think it's possible to remove duplication by moving this logic to methods on those structs?
```go
type configurations []*AppConfig
func (c configurations) humanString(w io.Writer) {
   for _, config := range c {
       fmt.Fprintf(w, "%s", ...)
   }
}

```